src/component/FilterSection.js

```diff
@@ -0,0 +1,95 @@
+import React from "react";
+import { useProduct } from "../context/product-context";
+import "../App.css";
+function FilterSection() {
+  const { state, dispatch } = useProduct();
+  console.log(state);
+  return (
+    <div className="filter-div">
+      <button
+        className="txt-lg bold-font clear-btn"
+        onClick={() => dispatch({ type: "RESET" })}
+      >
+        All Clear
+      </button>
+      <h1 className="txt-lg">Sort by Price</h1>
+      <div className="">
+        <label>High to low</label>
+        <input
+          type="radio"
+          name="SORT"
+          checked={state.sortByHighToLow}
+          value="HIGH_TO_LOW"
+          onClick={(e) => dispatch({ type: "SORT", payload: e.target.value })}
+        />
+      </div>
+      <div>
+        <label>low to High</label>
+        <input
+          type="radio"
+          name="SORT"
+          checked={state.sortByLowToHigh}
+          value="LOW_TO_HIGH"
+          onClick={(e) => dispatch({ type: "SORT", payload: e.target.value })}
+        />
+      </div>
+      <h1 className="txt-lg">Sort by Category</h1>
+      <div>
+        <label>Light Coffee</label>
+        <input
+          type="checkbox"
+          value="LIGHT_COFFEE"
+          checked={state.category.includes("LIGHT_COFFEE")}
+          onClick={(e) =>
+            dispatch({ type: "CATEGORY", payload: e.target.value })
```

Comment:
you are not handling the case when the user unchecks a particular category, this will only push the checked category to the state but neer remove it